### DISCUSSION TITLE
QueryMultipleAsync returns incorrect results if not all target class properties are matched

### DISCUSSION THREAD

**Discussion by nickjudson** [linq2db/linq2db#4743] [Q&A]
Perhaps this is by design, but if I have the following:

`db.QueryMultipleAsync<MyResult>("3 Select statements", parameters)`

which map to
```
public class MyResult
{
  public IList<Type1> Result1 {get; set;}
  public IList<Type2> Result2 {get; set;}
  public IList<Type3> Result3 {get; set;}
}
```

Then I see the expected results: Each property is filled in as expected.

However, if I remove one of the SQL statements from the SQL string (so only 2 result sets), then the results become incorrect. It would appear that the number of properties in the target class need to exactly match the number of SQL statements.

I'm using dynamically generated SQL, where I may return the same number of result sets as there are target class properties, but I may also just return a subset.

Is there a class/property attribute which needs to be used to account for this?

`Version:  5.4.1` against both SQL Server and PG. Same behavior for QueryProcMultipleAsync.

**Comment by MaceWindu** [comment]
There is `ResultSetIndexAttribute` attribute.
When it applied to properties of `MyResult` class - only properties with this attribute mapped in specified order. Otherwise all writeable properties are set in reflection order.

**Reply by nickjudson** [reply]
Interesting. I guess this won't work for dynamic SQL then as the order isn't determinate. For it to work, reflection would need to find the correct property based on type. Thanks for the reply!
